controllers/front/SmailyRssFeed.php

```diff
@@ -32,7 +32,33 @@ public function initContent()
 
     public function generateRssFeed()
     {
-        $products = Product::getProducts($this->context->language->id, 0, 50, 'date_upd', 'desc', false, true);
+        // Hardcoded values in v1.3.0 and older.
+        $limit = 50;
+        $order_by = 'date_upd';
+        $order_way = 'desc';
+        $id_category = false;
+        // Make sure the values are what we are expecting.
+        if (Tools::getValue('limit') >= 1 && is_int(Tools::getValue('limit'))) {
```

Comment:
I am sure this condition won't work, because in order for `is_int` to return a truthy value it needs an integer value passed to it, but query arguments are always of type `string` and `Tools::getValue()` does not do any special conversion logic.

```php
php > is_int('10');
bool(false)
php > is_int(10);
bool(true)
```

Reply:
Typecasting in new commit and comparing it